src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+                      [(and y-uflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x overflows and y rescues it
+                      [(and x-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; y overflows and x rescues it
+                      [(and y-oflow? (not (= (abs subexpr-val) +0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+
+                   [(list '*.f64 x-ex y-ex)
+                    #:when (or (is-inexact? x-ex) (is-inexact? y-ex))
+                    (define y-oflow? (overflow? y-ex))
+                    (define x-uflow? (underflow? x-ex))
+                    (define x-oflow? (overflow? x-ex))
+                    (define y-uflow? (underflow? y-ex))
+                    (cond
+                      ; 0 * inf ~ nan, but if they rescue each other, then it should not be nan
+                      [(and x-uflow? y-oflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; inf * 0 ~ nan, but if they rescue each other, then it should not be nan
+                      [(and x-oflow? y-uflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x underflows and y is normal, then if x != 0, then error
+                      [(and x-uflow? (not (= (abs subexpr-val) 0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x overflows and y is normal, then if x != inf, then error
+                      [(and x-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x is normal and y underflows, then if x != 0, then error
+                      [(and y-uflow? (not (= (abs subexpr-val) 0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x is normal and y overflows, then if x != inf, then error
+                      [(and y-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+                   
+                   [(list 'log.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+                    (define cond-num (abs (/ 1 subexpr-val)))
+                    (define arg-oflow? (overflow? arg))
+                    (define arg-uflow? (underflow? arg))
+                    (cond
+                      ; Condition number hallucination:
+                      ; Condition number is high when x = 1,
+                      ; but x is exactly 1, so there is no error
+                      [(and (= arg-val 1.0) (= subexpr-val 0.0)) #f]
```

Comment:
Confused by this… how do we know x is exactly 1 and not just close to 1?

Reply:
For log(f(x)), we have R(f(x)), so if we just check if R(f(x)) == 1.0, then we can say that f(x) is exactly 1.0 right?

Reply:
Definitely not! If R(f(x)) is 1.0 f(x) could be 1.0000000000000000001, for example.

Reply:
(This only works for 0 because 0 is special.)

Reply:
I think that is why I am also checking if subexpr-val (which is R(log(x))) is also 0 at the same time. I think both of them together imply that f(x) is exactly 1.0

Reply:
Oh, I see, yes. (I mean, maybe not quite exactly, you can imagine that if x ~= 1 + 1e-500, then R(x) = 1 and R(log(x)) = 0 just because the epsilon is so small. But I imagine that's rare? It would be really nice if we had a real exact/non-exact test but it doesn't seem like that's the main blocker to making progress.